akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala

```diff
@@ -189,4 +205,187 @@ class InteractionPatternsSpec extends TestKit with TypedAkkaSpecWithShutdown {
+      val drawer: ActorRef[GetWallet] = ctx.spawn(drawerBehavior, "drawer")
+
+      msg match {
+        case LeaveHome(who, respondTo) ⇒
+          ctx.spawn(prepareToLeaveHome(who, respondTo, keyCabinet, drawer), s"leaving-$who")
+          Behavior.same
+      }
+    }
+
+    def prepareToLeaveHome(
+      whoIsLeaving: String,
+      respondTo:    ActorRef[ReadyToLeaveHome],
+      keyCabinet:   ActorRef[GetKeys],
+      drawer:       ActorRef[GetWallet]): Behavior[NotUsed] =
+      // we don't _really_ care about the actor protocol here as nobody will send us
+      // messages except for responses to our queries, so we just accept any kind of message
+      // but narrow that to more limited types then we interact
+      Behaviors.deferred[AnyRef] { ctx ⇒
+        var wallet: Option[Wallet] = None
+        var keys: Option[Keys] = None
+
+        // we narrow the ActorRef type to any subtype of the actual type we accept
+        keyCabinet ! GetKeys(whoIsLeaving, ctx.self.narrow[Keys])
+        drawer ! GetWallet(whoIsLeaving, ctx.self.narrow[Wallet])
+
+        def nextBehavior: Behavior[AnyRef] =
+          (keys, wallet) match {
+            case (Some(w), Some(k)) ⇒
+              // we got both, "session" is completed!
+              respondTo ! ReadyToLeaveHome(whoIsLeaving, w, k)
+              Behavior.stopped
+
+            case _ ⇒
+              Behavior.same
+          }
+
+        Behaviors.immutable((ctx, msg) ⇒ {
+          msg match {
+            case w: Wallet ⇒
+              wallet = Some(w)
+              nextBehavior
+            case k: Keys ⇒
+              keys = Some(k)
+              nextBehavior
+            case _ ⇒
+              Behaviors.unhandled
+          }
+        })
+      }.narrow[NotUsed] // we don't let anyone else know we accept anything
+    // #per-session-child
+
+    val requestor = TestProbe[ReadyToLeaveHome]()
+
+    val home = spawn(homeBehavior, "home")
+    home ! LeaveHome("Bobby", requestor.ref)
+    requestor.expectMsg(ReadyToLeaveHome("Bobby", Keys(), Wallet()))
+  }
+
+  "contain a sample for ask from outside the actor system" in {
+    // #standalone-ask
+    trait CookieCommand {}
+    case class GiveMeCookies(replyTo: ActorRef[Cookies]) extends CookieCommand
+    case class Cookies(count: Int)
+    // #standalone-ask
+
+    // keep this out of the sample as it uses the testkit spawn
+    val cookieActorRef = spawn(Behaviors.immutable[GiveMeCookies] { (ctx, msg) ⇒
+      msg.replyTo ! Cookies(5)
+      Behaviors.same
+    })
+
+    // #standalone-ask
+
+    import akka.actor.typed.scaladsl.AskPattern._
+
+    // asking someone requires a timeout, if the timeout hits without response
+    // the ask is failed with a TimeoutException
+    implicit val timeout: Timeout = 3.seconds
+    // the response callback will be executed on this execution context
+    import system.executionContext
+
+    val result: Future[Cookies] = cookieActorRef ? GiveMeCookies
```

Comment:
I think this is too magic. I think we should show two versions, first uncurried and then curried. To make it clear the message should also have another param. 

```
case class GiveMeCookies(money: Int, replyTo: ActorRef[Cookies]) extends CookieCommand

val result: Future[Cookies] = cookieActorRef.ask(replyTo => GiveMeCookies(100, replyTo))
```

Then we can show the curried:
```
case class GiveMeCookies(money: Int)(replyTo: ActorRef[Cookies]) extends CookieCommand

val result: Future[Cookies] = cookieActorRef.ask(GiveMeCookies(100))
val result2: Future[Cookies] = cookieActorRef ? GiveMeCookies(100)
```

Reply:
I think I'd rather just show a lambda one, and then clever people can figure out all the N clever ways to do it.

Reply:
sounds good, since that shows what it is